ansible/roles/dashmate/tasks/main.yml

```diff
@@ -173,8 +173,53 @@
   changed_when: true
   when: dashmate_branch is defined
 
-# TODO: we should use status command to figure out if dashmate is running
-#   to get the status from the response
+- name: Get status
+  ansible.builtin.command: "{{ dashmate_cmd }} status --format=json"
+  become: true
+  become_user: dashmate
+  changed_when: true
+  args:
+    chdir: '{{ dashmate_cwd }}'
+  register: dashmate_status
+
+- name: Get service status
+  ansible.builtin.set_fact:
+    core_service_status: "{{ dashmate_status.stdout | from_json | json_query('core.serviceStatus') }}"
+    tenderdash_service_status: "{{ dashmate_status.stdout | from_json | json_query('platform.tenderdash.serviceStatus') }}"
+
+- name: Fail if dashmate node in inconsistent state
+  ansible.builtin.fail:
+    msg: Dashmate status returned invalid service status
+  when: >
+    (core_service_status != 'up' and core_service_status != 'stopped') or
+    (tenderdash_service_status != 'up' and tenderdash_service_status != 'stopped')
+
+# start platform when (core is up and platform is down)
```

Comment:
Don't we need to restart Core as well? What if dashmate code is updated or config is changed?

We can do restart logic smarter:
1. Images are changes: restart core if core image changed and restart platform if platform images are changed
2. dashmate config is changed
3. dashmate is changed

Reply:
Right, yesterday I had difficulty updating from Platform 0.24.2 to 0.24.4 because `docker compose restart` does not pull and use newer images, `docker compose up -d` is necessary for this. We definitely need a way of ensuring the necessary containers are restarted (or stopped, updated and started) when config changes. But changing the dashmate version probably shouldn't cause core to restart - this usually changes at the same time as the platform version only.

Reply:
The full logic must be:
* check services status
  * core is running
  * drive is running
* start ALL when (core is down and platform is down)
* start platform when (core is up and platform is down)
* determine what's need to be restarted:
  * any of platform images are updated: platform
  * core / sentinel images are updated: all
  * dashmate is changed: all
  * dashmate config is changed: all
* restart platform (platform is not down and platform needs to be restarted)
* restart all (all needs to be restarted)